pkg/cri/instrument/instrumented_service.go

```diff
@@ -95,13 +95,25 @@ func (in *instrumentedService) checkInitialized() error {
 // GRPC service request handlers should return error before server is fully
 // initialized.
 // NOTE(random-liu): All following functions MUST check initialized at the beginning.
-func (in *instrumentedAlphaService) checkInitialized() error {
+func (in *instrumentedAlphaService) checkInitialized(ctx context.Context) error {
+	in.emitUsageWarning(ctx)
 	if in.c.IsInitialized() {
 		return nil
 	}
 	return errors.New("server is not initialized yet")
 }
 
+// emitUsageWarning emits a warning when v1alpha2 cri-api is called.
+func (in *instrumentedAlphaService) emitUsageWarning(ctx context.Context) {
+	// Only emit the warning the first time an v1alpha2 api is called
+	in.emitWarning.Do(func() {
```

Comment:
this LGTM. I don't know how big this problem will be for the end user - it may be interesting to log the specific call that was called. This way customers received this warning may understand who called it